elf_reader.go

```diff
@@ -585,6 +587,40 @@ func (ec *elfCode) relocateInstruction(ins *asm.Instruction, rel elf.Symbol) err
 			return fmt.Errorf("asm relocation: %s: unsupported type %s", name, typ)
 		}
 
+		// If a Call instruction is found and the datasec has a btf.Func with a Name
+		// that matches the symbol name we mark the instruction as a call to a kfunc.
+		if ins.OpCode.JumpOp() == asm.Call {
+			if ec.btf == nil {
+				break
+			}
+			var ds *btf.Datasec
```

Comment:
Can you refactor this into a method on `elfCode`? `elfCode.findKfunc(name string) (btf.Type, error)` or similar. That function can return `(nil, nil)` to indicate "no BTF or not a kfunc call".

Patching of ins would still happen here though.

Reply:
I opted for the `elfCode.loadKsymsSection()` which fills up the ksyms map on `elfCode`, that should go in a similar direction as you suggested I hope.